dev/io.openliberty.security.oidcclientcore.internal/src/io/openliberty/security/oidcclientcore/userinfo/UserInfoResponse.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.security.oidcclientcore.userinfo;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.ibm.json.java.JSONObject;
+
+public class UserInfoResponse {
+
+    private final JSONObject rawResponse;
+    private Map<String, String> responseAsMap;
+
+    public UserInfoResponse(JSONObject responseStr) {
+        rawResponse = responseStr;
+    }
+
+    @SuppressWarnings("unchecked")
+    public Map<String, String> asMap() {
+        if (responseAsMap != null) {
+            return responseAsMap;
+        }
+        if (rawResponse == null) {
+            return null;
+        }
+        Map<String, String> map = new HashMap<>();
+        Set<String> keys = rawResponse.keySet();
+        for (String key : keys) {
+            map.put(key, rawResponse.get(key).toString());
```

Comment:
```suggestion
            map.put(key, rawResponse.get(key));
```
We may want to make this a `Map<String, Object>` to avoid an issue like we've just seen with EC JWKs. The JSON string might contain things other than simple strings as values, so adding them as `Object`s here should help preserve their type if needed later.